### ISSUE TITLE
0.28.0 Routes/DNS broken after sleep on macOS

### ISSUE THREAD

**Issue by hurricanehrndz** [netbirdio/netbird#2093]
**Describe the problem**

On version 0.28.0, with networkmonitor on routes and dns fail after sleep

**To Reproduce**

1. Turn network monitor on in config
2. Sleep macbook for 10 mins
3. Wake
4. run yelpvpn status -d
5. note that DNS servers failed with i/o timeout
6. note that routed networks are unreachable although assigned to a routing peer, and in routing table

**Expected behavior**

Expect routed networks and DNS to be reachable after sleep

**Are you using NetBird Cloud?**

No

**NetBird version**
0.28.0

**NetBird status -d output:**

If applicable, add the `netbird status -d' command output.

![Image 6-4-24 at 10 01 AM](https://github.com/netbirdio/netbird/assets/5804237/dfec7774-d227-4721-b8b5-146dbcf883b4)

**Comment by hurricanehrndz** [comment]
It seems like with version 0.28.0 ACLs are missing when engine is restarted
```
peer: DdiC7cXVeE40xxx
  endpoint: 1xx.98.xxxx:51820
  allowed ips: 100.113.20.218/32
  latest handshake: 43 seconds ago
  transfer: 1.76 KiB received, 4.29 KiB sent
  persistent keepalive: every 25 seconds

peer: S3TburlUFqMPxxxx
  endpoint: 1x.xx.xx.27:51820
  allowed ips: 100.113.35.69/32
  latest handshake: 1 minute, 27 seconds ago
  transfer: 1.13 KiB received, 4.49 KiB sent
  persistent keepalive: every 25 seconds

peer: MGJ+hGFni8xxxx
  endpoint: 18x..x.x.x:51820
  allowed ips: 100.113.74.116/32
  latest handshake: 1 minute, 51 seconds ago
  transfer: 4.20 KiB received, 7.73 KiB sent
  persistent keepalive: every 25 seconds

```

**Comment by hurricanehrndz** [comment]
Confirmed in the logs on version 0.28.0
```
24870:2024-06-05T14:10:13-06:00 INFO client/internal/routemanager/client.go:175: New chosen route is co6ojai9l7bha9qbpge0:co6mgqa9l7bha9qbpf70 with peer S3TburlUFqMPX3PLDlKhQnD/NLxq9B9YCzJ7mAZozh8= with score 99002.937566 for network [172.19.32.0/22]
25321:2024-06-05T14:10:13-06:00 TRAC client/internal/routemanager/refcounter/refcounter.go:54: Increasing ref count 0 for prefix 172.19.32.0/22
25322:2024-06-05T14:10:13-06:00 TRAC client/internal/routemanager/refcounter/refcounter.go:58: Adding for prefix 172.19.32.0/22
25343:2024-06-05T14:10:13-06:00 TRAC client/internal/routemanager/systemops/systemops_unix.go:64: route -n add -inet 172.19.32.0/22 -interface utun4319: add net 172.19.32.0: gateway utun4319
25345:2024-06-05T14:10:13-06:00 TRAC client/internal/routemanager/refcounter/refcounter.go:54: Increasing ref count 3 for prefix 172.19.32.0/22
```
with a new chosen route there is entry for adding the allowed network to the interface

version 0.27.x
```
client.log:29358:2024-06-05T13:13:02-07:00 INFO client/internal/routemanager/client.go:165: new chosen route is co6ojai9l7bha9qbpge0:co6mgqa9l7bha9qbpf70 with peer S3TburlUFqMPX3PLDlKhQnD/NLxq9B9YCzJ7mAZozh8= with score 99002.932073 for network 172.19.32.0/22
client.log:29856:2024-06-05T13:13:02-07:00 DEBG iface/iface.go:106: removing allowed IP from interface utun4319 and peer DdiC7cXVeE409wdJO1bQjFQ8JYP52X+/HO46FtkTQSI=: allowed IP 172.19.32.0/22 
client.log:30379:2024-06-05T13:13:02-07:00 DEBG iface/iface.go:97: adding allowed IP to interface utun4319 and peer S3TburlUFqMPX3PLDlKhQnD/NLxq9B9YCzJ7mAZozh8=: allowed IP 172.19.32.0/22 
```

**Comment by hurricanehrndz** [comment]
Seems like this function never gets the changes applied to the wg iface
https://github.com/netbirdio/netbird/pull/2092/files#diff-0712adc281c7830f45e8a4035fad97f4d803abf05a094798a6734e8c51f3ee5fR39

**Comment by mlsmaycon** [comment]
Thank you for reporting the issue @hurricanehrndz we will check it first thing tomorrow

**Comment by lixmal** [comment]
@hurricanehrndz can you test the `0.28.0` branch with the merged PR?

**Comment by hurricanehrndz** [comment]
Will do

**Comment by hurricanehrndz** [comment]
#2099 fixes  issue #2093

**Comment by mlsmaycon** [comment]
We are targeting next Thursday as release date. We are still testing it out and there are more features to come too.
